backend/wmg/data/transform.py

```diff
@@ -38,70 +38,174 @@ def get_cell_types_by_tissue(corpus_group: str) -> Dict:
-        ancestors = [i for s in ancestors for i in s]
-        ancestors = set(ancestors)
+    G = pgv.AGraph()
+    for a in ancestors:
+        for s in a.subclasses(with_self=False, distance=1):
+            if s in ancestors:
+                G.add_edge(a.id, s.id)
 
-        G = pgv.AGraph()
-        for a in ancestors:
-            for s in a.subclasses(with_self=False, distance=1):
-                if s in ancestors:
-                    G.add_edge(a.id, s.id)
+    G.layout(prog="dot")
 
-        G.layout(prog="dot")
+    positions = {}
+    for n in G.iternodes():
+        pos = n.attr["pos"].split(",")
+        positions[n] = (float(pos[0]), float(pos[1]))
 
-        positions = {}
-        for n in G.iternodes():
-            pos = n.attr["pos"].split(",")
-            positions[n] = (float(pos[0]), float(pos[1]))
+    ancestor_ids = [a.id for a in ancestors]
 
-        ancestor_ids = [a.id for a in ancestors]
+    def cell_entity(node, depth):
+        return {"id": node, "depth": depth}
 
-        def recurse(node: Set[str], depth=0):
+    def recurse(node: str, depth=0):
 
-            if node in cells:
+        if node in cells:
 
-                cells.remove(node)
-                yield {"id": node, "depth": depth}
+            cells.remove(node)
+            yield cell_entity(node, depth)
 
-                if node != "CL:0000003":
-                    depth += 1
+            # Make the root and other high level cell types stay at 0 level. Avoids creating
+            # a situation where only the root is at lowest level
+            if node not in ["CL:0000003", "CL:0000255", "CL:0000548"]:
+                depth += 1
 
-            children = [
-                (c, positions[c.id]) for c in onto[node].subclasses(with_self=False, distance=1) if c.id in ancestor_ids
-            ]
-            sorted_children = sorted(children, key=lambda x: x[1][0])
-            for child in sorted_children:
-                yield from recurse(child[0].id, depth=depth)
+        children = [
+            (c, positions[c.id]) for c in onto[node].subclasses(with_self=False, distance=1) if c.id in ancestor_ids
+        ]
+        sorted_children = sorted(children, key=lambda x: x[1][0])
+        for child in sorted_children:
+            yield from recurse(child[0].id, depth=depth)
 
-        ordered_list = recurse(root)
-        return list(ordered_list)
+    # Apply recursion to create ordered list of cells present in set "cells"
+    ordered_list = list(recurse(root))
 
-    mapping = {}
-    for tissue, cell_df in cell_type_by_tissue.items():
-        cells = list(cell_df)
-        ordered_cells = compute_ordering(cells, "CL:0000003")
-        mapping[tissue] = ordered_cells
+    # If there are any cells left in set "cells", it means that either those cell types
+    # don't exist in the ontology or they are above the root ("CL:0000003")
+    # Add these "orphan" cells at end of list
+    while cells:
```

Comment:
You can use: `ordered_list.extend([cell_entity(cell) for cell in cells])`, unless I am missing something.